analytic_engine/src/lib.rs

```diff
@@ -166,3 +194,20 @@ impl Default for Config {
         }
     }
 }
+
+/// Analytic dynamic config
+#[derive(Debug, Default, Clone)]
+pub struct DynamicConfig {
```

Comment:
I guess this is not a good way to provide the feature for dynamic configurations because the details of the dynamic config has been exposed and any changes of it will lead to the caller's code update. So how about exposing a trait or struct called `DynamicConfigger` to provide specific method to do config dynamic change?